protocol/chainlib/grpc.go

```diff
@@ -200,11 +200,12 @@ func (apil *GrpcChainListener) Serve(ctx context.Context) {
 		var relayReply *pairingtypes.RelayReply
 		metricsData := metrics.NewRelayAnalytics("NoDappID", apil.endpoint.ChainID, apiInterface)
 		if relayReply, _, err = apil.relaySender.SendRelay(ctx, method, string(reqBody), "", "NoDappID", metricsData); err != nil {
-			go apil.logger.AddMetric(metricsData, err != nil)
+			go apil.logger.AddMetricForGrpc(metricsData, false, ctx)
```

Comment:
can move this to right after sendRelay (before error handling) 
and parse the error inside, do this to all API's 

go apil.logger.AddMetricForGrpc(metricsData, err, ctx)

that way we can call it only once

Reply:
saw you did that for the other api's so here as well maybe you forgot

Reply:
do you mean to change the code like that?
```
relayReply, _, err = apil.relaySender.SendRelay(...)
go apil.logger.AddMetric(metricsData, err != nil)
if err != nill {
...
}
```

Reply:
nope i didn't forget, the code has a return inside the if (line 202)